### ISSUE TITLE
Mixing static role with LDAP query may hide it

### ISSUE THREAD

**Issue by frafra** [dalibo/ldap2pg#356]
## What do you want?

This configuration works with ldap2pg 5.0 and 5.1, but not with  ldap2pg 5.2 and higher. I get the following error, but I do not understand what to do.

`Failed to load configuration: Mixing static role with LDAP query may hide it.`

## ldap2pg.yml

```yml
verbosity: 5
color: yes
dry: no

postgres:
  dsn: postgresql://postgres@localhost/postgres
  blacklist: [postgres, root, admin, zabbix, commvault, public, pg_*]
  databases_query: |
    SELECT datname
    FROM pg_catalog.pg_database
    WHERE datallowconn IS TRUE
    AND NOT datname='postgres'
    AND NOT datname='template1';
ldap:
  uri: ldap://ldap.mycompany.com
  binddn: ldapquery@mycompany.com
  password: fancypassword

privileges:
  ro:
  - __connect__
  - __usage_on_schemas__
  - __select_on_tables__
  rw:
  - ro
  - __temporary__
  - __insert__
  - __update_on_tables__
  admin:
  - rw
  - __all_on_schemas__
  - __all_on_tables__

sync_map:
- roles:
  - name: mycompany_users
    options: NOLOGIN
    comment: Default grp for all AD synced users

- ldap:
    base: OU=Brukere,DC=mycompany,DC=no
    scope: sub
    filter: >
      (&
        (objectClass=User)
        (sAMAccountName=*.*)
        (memberOf:1.2.840.113556.1.4.1941:=cn=ag_pgsql_superuser,OU=PostgreSQL,OU=Access,OU=Groups,DC=mycompany,DC=no)
      )
  roles:
  - names:
    - '{sAMAccountName}'
    options: LOGIN SUPERUSER INHERIT CREATEDB CREATEROLE REPLICATION
    parents:
    - mycompany_users
    comment: User synced from mycompany AD
- ldap:
    base: OU=Brukere,DC=mycompany,DC=no
    scope: sub
    filter: >
      (&
        (objectClass=User)
        (sAMAccountName=*.*)
        (memberOf:1.2.840.113556.1.4.1941:=cn=ag_pgsql_project_admin,OU=PostgreSQL,OU=Access,OU=Groups,DC=mycompany,DC=no)
      )
  roles:
  - names:
    - '{sAMAccountName}'
    options: LOGIN
    parents:
    - mycompany_users
    - ag_pgsql_project_admin
    comment: User synced from mycompany AD
  grant:
    database: postgres
    privilege: admin
    schema: __all__
    role: ag_pgsql_project_admin

- ldap:
    base: OU=Brukere,DC=mycompany,DC=no
    scope: sub
    filter: >
      (&
        (objectClass=User)
        (sAMAccountName=*.*)
        (memberOf:1.2.840.113556.1.4.1941:=cn=ag_pgsql_project_rw,OU=PostgreSQL,OU=Access,OU=Groups,DC=mycompany,DC=no)
      )
  roles:
  - names:
    - '{sAMAccountName}'
    options: LOGIN
    parents:
    - mycompany_users
    - ag_pgsql_project_rw
    comment: User synced from mycompany AD
  grant:
    database: postgres
    privilege: rw
    schema: __all__
    role: ag_pgsql_project_rw
- ldap:
    base: OU=Brukere,DC=mycompany,DC=no
    scope: sub
    filter: >
      (&
        (objectClass=User)
        (sAMAccountName=*.*)
        (memberOf:1.2.840.113556.1.4.1941:=cn=ag_pgsql_project_ro,OU=PostgreSQL,OU=Access,OU=Groups,DC=mycompany,DC=no)
      )
  roles:
  - names:
    - '{sAMAccountName}'
    options: LOGIN
    parents:
    - mycompany_users
    - ag_pgsql_project_ro
    comment: User synced from mycompany AD
  grant:
    database: postgres
    privilege: ro
    schema: __all__
    role: ag_pgsql_project_ro
```

**Comment by bersace** [comment]
Hi @frafra.

Yes, ldap2pg 5.2 added this check. This avoid dropping static roles if ldap queries returns no entries.

I'm thinking of another way to deal with this, by extracting the static rule from the ldap rule after parsing the configuration.

**Comment by bersace** [comment]
Fixed in ldap2pg 5.6.
